bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -202,36 +194,36 @@ impl Client {
         });
 
         if let Some(process_id) = cross_process_refresh_lock_id {
-            let session_delegate = client
-                .session_delegate
-                .clone()
-                .context("missing session delegates when enabling the cross-process lock")?;
-
+            if session_delegate.is_none() {
+                return Err(anyhow::anyhow!(
+                    "missing session delegates when enabling the cross-process lock"
+                ))?;
+            }
```

Comment:
Why not use `.context()` here as before?

Reply:
We're not doing anything with the `session_delegate` so we'd need to use `anyhow::ensure` or `anyhow::bail`, but this function returns `ClientError` so we need a way to convert into that, which the `?` achieves.

Reply:
Oh, the `return` does nothing. I think you could have used `.context()?` still though, don't see why not. You don't have to use the result of that.